src/js/letters/components/Address.jsx

```diff
@@ -101,74 +95,84 @@ class Address extends React.Component {
+          onValueChange={(update) => this.props.onInput('addressOne', update)}/>
         <ErrorableTextInput
           label="Street address (optional)"
           name="address"
           autocomplete="street-address"
           charMax={30}
           value={this.props.address.addressTwo}
-          onValueChange={(update) => {this.props.onInput('addressTwo', update);}}/>
+          onValueChange={(update) => this.props.onInput('addressTwo', update)}/>
         <ErrorableTextInput
           label="Street address (optional)"
           name="address"
           autocomplete="street-address"
           charMax={30}
           value={this.props.address.addressThree}
-          onValueChange={(update) => {this.props.onInput('addressThree', update);}}/>
-        <ErrorableTextInput errorMessage={this.isValidAddressField(this.props.address.city) ? undefined : 'Please enter a city'}
+          onValueChange={(update) => this.props.onInput('addressThree', update)}/>
+        <ErrorableTextInput errorMessage={errorMessages.city}
           label={<span>City <em>(or APO/FPO/DPO)</em></span>}
           name="city"
           autocomplete="address-level2"
           charMax={30}
           value={this.props.address.city}
           required={this.props.required}
-          onValueChange={(update) => {this.props.onInput('city', update);}}/>
-        {stateProvince}
-        <ErrorableTextInput errorMessage={this.isValidPostalCode(this.props.address.zipCode) ? undefined : 'Please enter a valid Postal code'}
+          onValueChange={(update) => this.props.onInput('city', update)}/>
+        <ErrorableSelect errorMessage={errorMessages.state}
+          label="State"
+          name="state"
+          autocomplete="address-level1"
+          options={adjustedStateNames}
+          value={this.props.address.state}
+          required={this.props.required}
+          onValueChange={(update) => this.props.onInput('state', update)}/>)
+
+        {/* Hide the zip code for addresseses that aren't in the US */}
+        {selectedCountry === 'USA' && <ErrorableTextInput errorMessage={errorMessages.zipCode}
           additionalClass="usa-input-medium"
-          label={selectedCountry === 'USA' ? 'Zip code' : 'Postal code'}
+          label={'Zip code'}
           name="postalCode"
           autocomplete="postal-code"
           value={this.props.address.zipCode}
           required={this.props.required}
-          onValueChange={(update) => {this.props.onInput('zipCode', update);}}/>
+          onValueChange={(update) => this.props.onInput('zipCode', update)}/>
+        }
       </div>
     );
   }
 }
 
+const addressShape = PropTypes.shape({
+  addressOne: PropTypes.string,
+  addressTwo: PropTypes.string,
+  addressThree: PropTypes.string,
+  city: PropTypes.string,
+  state: PropTypes.string,
+  country: PropTypes.string,
+  zipCode: PropTypes.string
+});
+
+Address.propTypes = {
+  // value = address
```

Comment:
Think you can remove this?

Reply:
Ah! Yes. Because now `this.props.value` actually _is_ `this.props.address`. Good catch!